src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -0,0 +1,360 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Security;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
+
+using SafeWinHttpHandle = Interop.WinHttp.SafeWinHttpHandle;
+
+namespace System.Net.Http
+{
+    /// <summary>
+    /// Static class containing the WinHttp global callback and associated routines.
+    /// </summary>
+    internal static class WinHttpRequestCallback
+    {
+        public static Interop.WinHttp.WINHTTP_STATUS_CALLBACK s_StaticCallbackDelegate =
```

Comment:
I don't think I fully understood how this works: What keeps the callback from getting GCed before the SafeHandles actually close. 
On handle close WinHttp will do one final callback. If the callback object was disposed, CLR will trigger an MDA in debug mode or a crash in a ret build.